core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -421,6 +421,13 @@ object SparkHadoopUtil {
 
   val SPARK_YARN_CREDS_COUNTER_DELIM = "-"
 
+  // Just load HdfsConfiguration into the class loader to add
+  // hdfs-site.xml as a default configuration file otherwise
+  // some HDFS related configurations doesn't ship to Executors and
+  // it can cause UnknownHostException when NameNode HA is enabled.
+  // See SPARK-11227 for more details.
+  Utils.classForName("org.apache.hadoop.hdfs.HdfsConfiguration")
```

Comment:
As you mentioned, Hadoop 2.2 have `HdfsConfiguration` but are there any way to load `HdfsConfiguration` explicitly without reflection?